### PR TITLE
[REF] Add in function to retrieve Pre Upgrade messages in headless

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This makes it possible for CLI tools such as drush and wp cli to get the pre-upgrade message when doing database upgrades

Before
----------------------------------------
No way to get the pre upgrade messages just post

After
----------------------------------------
function to get the pre-upgrade message

ping @demeritcowboy @kcristiano @totten

### CODE REVIEW

**Comment by totten** [comment]
civibot, test this please

**Comment by totten** [comment]
Looks sensible. @seamuslee001 Are you working on a patch for a tool/entry-point that can be used to `r-run` it?

File: CRM/Upgrade/Headless.php [hunk 1]
```diff
@@ -38,8 +44,11 @@ public function run($enablePrint = TRUE) {
     CRM_Core_DAO::dropTriggers();
 
     // CRM-11156
-    $preUpgradeMessage = NULL;
-    $upgrade->setPreUpgradeMessage($preUpgradeMessage, $currentVer, $latestVer);
+    if (empty($this->preUpgradeMessage)) {
+      $preUpgradeMessage = NULL;
+      $upgrade->setPreUpgradeMessage($preUpgradeMessage, $currentVer, $latestVer);
```
Comment (eileenmcnaughton) [inline CRM/Upgrade/Headless.php:49]: Ug - this is a bit nasty that it is a pass-by-ref huh? Doesn't look like it is trivial to unravel though

Comment (seamuslee001) [reply CRM/Upgrade/Headless.php:49]: It is pass by reference

**Comment by seamuslee001** [comment]
@totten see https://github.com/civicrm/civicrm-drupal/pull/663 I plan to do one for wp-cli as well

File: CRM/Upgrade/Headless.php (hunk 1 above)
Comment (totten) [reply CRM/Upgrade/Headless.php:49]: Yeah, it's also a confusing name... but there are 3 distinct structures which include variations on this signature (`CRM_Upgrade_Incremental_Base` and its subclasses; then also `CRM_Upgrade_Form`; then also `CRM_Upgrade_Incremental_General`).

**Comment by totten** [comment]
Did an `r-run` using this PR and the accompanying drupal PR. It succeeded in get the pre-upgrade messages, so let's merge.

**Comment by totten** [comment]
I pulled the merge trigger a bit hastily. Suggested revision: https://github.com/civicrm/civicrm-core/pull/25488

**Pull request merged**